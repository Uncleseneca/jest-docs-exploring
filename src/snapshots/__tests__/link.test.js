import React from 'react';
import Link from '../Link';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<Link page="vk.com">vk</Link>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('changes class when hovered', () => {
  const component = renderer.create(<Link page="google.com">google</Link>);

  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  // trigger callback manually
  tree.props.onMouseEnter();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onMouseLeave();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
