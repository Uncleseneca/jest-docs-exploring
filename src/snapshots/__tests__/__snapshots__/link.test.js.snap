// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes class when hovered 1`] = `
<a
  className="normal"
  href="google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  google
</a>
`;

exports[`changes class when hovered 2`] = `
<a
  className="hovered"
  href="google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  google
</a>
`;

exports[`changes class when hovered 3`] = `
<a
  className="normal"
  href="google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  google
</a>
`;

exports[`renders correctly 1`] = `
<a
  className="normal"
  href="vk.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  vk
</a>
`;
